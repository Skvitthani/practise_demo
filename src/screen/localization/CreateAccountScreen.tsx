import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';
import {NavigationType} from '../../navigation';
import StringConst from '../../utils/StringConst';
import {View, Text, StyleSheet} from 'react-native';
import ButtonComp from '../../components/ButtonComp';
import InputTextComp from '../../components/InputTextComp';
import {NativeStackScreenProps} from '@react-navigation/native-stack';

type CreateAccountScreen = NativeStackScreenProps<
  NavigationType,
  'CreateAccountScreen'
> & {
  navigation: () => void;
};

const CreateAccountScreen: React.FC<CreateAccountScreen> = ({navigation}) => {
  const [name, setName] = useState<string | number>('');
  const [email, setEmail] = useState<string | number>('');
  const [password, setPassword] = useState<string | number>('');
  const {t} = useTranslation();

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{t(StringConst.Sign_up)}</Text>
      <InputTextComp
        value={name}
        placeholder={t(StringConst.Enter_Name)}
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setName(txt)}
      />
      <InputTextComp
        value={email}
        placeholder={t(StringConst.Enter_Email_Id_Name)}
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setEmail(txt)}
      />
      <InputTextComp
        value={password}
        placeholder={t(StringConst.Enter_Password)}
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setPassword(txt)}
      />

      <View style={styles.loginBtn}>
        <ButtonComp
          title={t(StringConst.Sign_up)}
          onPress={() => navigation.goBack()}
        />
      </View>
    </View>
  );
};

export default CreateAccountScreen;
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    fontSize: 20,
    color: '#000',
    marginTop: 100,
    fontWeight: '800',
    alignSelf: 'center',
  },
  inputStyle: {
    height: 50,
    width: '90%',
    marginTop: 30,
    paddingLeft: 20,
    borderWidth: 0.5,
    borderRadius: 10,
    alignSelf: 'center',
  },
  loginBtn: {
    marginTop: 50,
  },
});
