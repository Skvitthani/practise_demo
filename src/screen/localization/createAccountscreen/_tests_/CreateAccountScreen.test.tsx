import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import CreateAccountScreen from '../CreateAccountScreen';

jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => ({
    t: jest.fn(key => key),
  }),
}));

const component = (
  <CreateAccountScreen
    navigation={
      {navigate: jest.fn(), goBack: jest.fn(), canGoBack: jest.fn()} as any
    }
    route={undefined as any}
  />
);

describe('CreateAccountscreen', () => {
  test('renders snap', () => {
    const {toJSON} = render(component);
    expect(toJSON()).toMatchSnapshot();
  });
  it('name text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('name_text_input');
    fireEvent.changeText(container);
  });
  it('password text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('password_text_input');
    fireEvent.changeText(container);
  });
  it('email text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('email_text_input');
    fireEvent.changeText(container);
  });
  it('sign up button', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('sign_up_button');
    fireEvent.press(container);
  });
});
