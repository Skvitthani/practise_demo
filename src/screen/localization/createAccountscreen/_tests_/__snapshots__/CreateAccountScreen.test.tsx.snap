// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountscreen renders snap 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#000",
        "fontSize": 20,
        "fontWeight": "800",
        "marginTop": 100,
      }
    }
  >
    Sign up
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Name"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "height": 50,
          "marginTop": 30,
          "paddingLeft": 20,
          "width": "90%",
        },
      ]
    }
    testID="name_text_input"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Email Id Name"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "height": 50,
          "marginTop": 30,
          "paddingLeft": 20,
          "width": "90%",
        },
      ]
    }
    testID="email_text_input"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Password"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "height": 50,
          "marginTop": 30,
          "paddingLeft": 20,
          "width": "90%",
        },
      ]
    }
    testID="password_text_input"
    value=""
  />
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706233",
          "borderRadius": 20,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="sign_up_button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Sign up
      </Text>
    </View>
  </View>
</View>
`;
